### ISSUE TITLE
Possible minor error in nonparametric.py

### ISSUE THREAD

**Issue by RussoMarioDamiano** [kn27/BNPF#1]
Hey, stumbled upon your code while writing my Masters thesis (focusing on Poisson Factorization). First of all, kudos for the code, it was really helpful in figuring out the BNPPF model which is by no means trivial.

I think there might be a minor error at line 109 of "nonparametric.py": since you are working with $E[log \theta_{u, T+1}]$ I believe you should replace:
`np.sum(np.log(1 - self._v[u,:self.T]))`
with:
`np.sum(np.log(1 - self._v[u,:(self.T+1)]))`
as the product of the stick-breaking process occurs from 1 to (T+1)-1=T and not to T-1.

In practice, this however should have little effect on the convergence of variational parameters since it implies a small change in value in a small value (the infinite-sum). 

Cheers!

**Comment by kn27** [comment]
Hi there,

I did this for a class last year, thanks for commenting! I did not manage
to replicate the results from Gopalan, the converged results were inferior
to the parametric counterpart, if I remember correctly. If you happen to
work on the same paper and get the code working, I would be eager to hear
from you :)

Cheers,
Keane

**Comment by RussoMarioDamiano** [comment]
Yes, I managed to implement both models and got the nonparametric model slightly underperforming the parametric one. 
The reason is that the parametric model we both implemented is "**Hierarchical** Poisson Factorization", which is a 2014 revision of the 2013 "Poisson Factorization" model by Gopalan. (See that there are 3 versions of the Poisson Factorization model at: https://arxiv.org/abs/1311.1704v2). The difference between the parametric hierarchical and the vanilla parametric model lays in the added hierarchical level of user activities $\xi_u$ and item popularities $\eta_i$ which increase model flexibility and better capture the heterogeneity in user behavior.

The Nonparametric model is based on the non-hierarchical one, hence the lower performance due to the missing added layer. It would be nice to see the performance of a hierarchical nonparametric model, although this would make derivation of the update rules very hard if not impossible.

Hope it helps!
Best,
Mario
